keanu-project/src/test/java/io/improbable/keanu/vertices/AssertVertexTest.java

```diff
@@ -0,0 +1,88 @@
+package io.improbable.keanu.vertices;
+
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.algorithms.variational.optimizer.KeanuOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.ConstantBoolVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class AssertVertexTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void assertThrowsOnFalseConstBool() {
+        thrown.expect(AssertionError.class);
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(false));
+        constBool.assertTrue().eval();
+    }
+
+    @Test
+    public void assertPassesOnTrueConstBool() {
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(true));
+        constBool.assertTrue().eval();
+    }
+
+    @Test
+    public void lazyEvalThrowsOnFalseConstBool() {
+        thrown.expect(AssertionError.class);
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(false));
+        constBool.assertTrue().lazyEval();
+    }
+
+    @Test
+    public void assertPassesOnRVWithTruePredicate() {
+        UniformVertex uniform = new UniformVertex(0,5);
+        BoolVertex predicate = uniform.lessThan(new ConstantDoubleVertex(new double[]{10}));
+        predicate.assertTrue().eval();
+    }
+
+    @Test
+    public void assertPassesOnRVWithFalsePredicate() {
+        thrown.expect(AssertionError.class);
+        UniformVertex uniform = new UniformVertex(0,5);
+        BoolVertex predicate = uniform.greaterThan(new ConstantDoubleVertex(new double[]{10}));
+        predicate.assertTrue().eval();
+    }
+
+    @Test
+    public void samplingWithAssertionWorks() {
+        thrown.expect(AssertionError.class);
+        GaussianVertex uniform = new GaussianVertex(5,1);
```

Comment:
rename this to `gaussian` not `uniform`. Why do we need the `observingVertex`? I don't really understand what it adds to this test